src/index.js

```diff
@@ -11,7 +11,7 @@ const messages = utils.ruleMessages(ruleName, {
   expected,
 })
 const reSkipProp = /^(?:@|\$|--).+$/
-const reVar = /^-?(?:@.+|\$.+|var\(--.+\))$/
+const reVar = /^-?(?:@.+|(?:[^(]+\.)?\$.+|var\(--.+\))$/
```

Comment:
This pattern works but it has the potential to introduce false positives due to accepting "any characters". For example:

```css
.test::before {
  content: ".$5";
}
```

I’m not sure how much it matters considering how unlikely it is for a CSS value to contain `.$` 🤔.

From what I could find [in the spec](https://github.com/sass/sass/blob/master/accepted/module-system.md#member-references) it looks like the namespace is meant to be a valid CSS [ident-token](https://drafts.csswg.org/css-syntax-3/#ident-token-diagram), so a more bulletproof regex might be:

```regex
/^-?(?:@.+|(?:[a-zA-Z0-9-_]+\.)?\$.+|var\(--.+\))$/
```

This is more permissive than the actual definition of ident-token from the spec, but that shouldn’t really be a problem for the purpose of this plugin.

Reply:
Thank you for your instant feedback 👍 
You are right, all patterns in this plugin are loose and very permissive.
So far a duck on purpose^^

`content` rule is a very good point 👍 
I should extend all tests with that rule.

Anyway as you said, it leads to false positives.
Therefor a proper CSS properties/values parser would be the best solution.

I looked already into `postcss-values-parser`, which supports atm only standard CSS Syntax, so Sass and Less is still missing.
Though changing that would be a major release.